unit_tests/R7LoggerTests.php

```diff
@@ -86,7 +86,7 @@ public function testGetAddress()
 		public function testEmptyToken() {
 			$this->expectException(InvalidArgumentException::class);
 
-			R7Logger::getLogger($this->definitely_a_valid_attribute, self::region, self::persistent, true, self::severity, self::datahubEnabled, self::datahubIPAddress, self::datahubPort, self::host_id, self::host_name, self::host_name_enabled, self::add_local_timestamp, self::use_json);
+			R7Logger::getLogger("", self::region, self::persistent, true, self::severity, self::datahubEnabled, self::datahubIPAddress, self::datahubPort, self::host_id, self::host_name, self::host_name_enabled, self::add_local_timestamp, self::use_json);
```

Comment:
not sure if intentional, but this changes the test case from NULL to an empty string.

Reply:
Intentional, travis seems to crash on undefined attribute but my local doesn't.
Also this tests afaik tests the empty check we have, empty evaluates to true on 0, `""` so on so this to me is still a valid check
https://github.com/sbialkowski-r7/r7insight_php/blob/composer-add-readme/R7Logger.php#L155

Reply:
ah, yes. let's develop a language that has a settings file that allows users to change the behaviour of the entire language. sounds like a wonderful idea.

Reply:
as for equality checks; https://www.php.net/manual/en/types.comparisons.php#types.comparisions-loose
it depends on the operator used.